### DISCUSSION TITLE
How to install and use Clair?

### DISCUSSION THREAD

**Discussion by inf3790** [quay/clair#1664] [Q&A] [answered]
Hallo, I'm pretty new to Clair. I cloned the repository and used `docker-compose up`.
 After that with `docker ps` my output: 
`CONTAINER ID   IMAGE                             COMMAND                  CREATED             STATUS                   PORTS                                                                                                                                                                                    NAMES
2e7bd91d2bf7   traefik:v2.2                      "/entrypoint.sh trae…"   About an hour ago   Up 7 minutes             0.0.0.0:6060->6060/tcp, :::6060->6060/tcp, 80/tcp, 0.0.0.0:8080->8080/tcp, :::8080->8080/tcp, 0.0.0.0:49158->5432/tcp, :::49158->5432/tcp, 0.0.0.0:49157->8443/tcp, :::49157->8443/tcp   clair-traefik
184dc378394d   quay.io/projectquay/golang:1.17   "go run . -conf /etc…"   About an hour ago   Up 7 minutes                                                                                                                                                                                                      clair-matcher
9d91c16efb2a   quay.io/projectquay/golang:1.17   "go run . -conf /etc…"   About an hour ago   Up 7 minutes                                                                                                                                                                                                      clair-indexer
b23b33fff26e   postgres:12                       "docker-entrypoint.s…"   About an hour ago   Up 7 minutes (healthy)   5432/tcp                                                                                                                                                                                 clair-database`

Now I want to submit a manifest, for that I need clairctl. If I try to follow the documentation [is](https://quay.github.io/clair/howto/getting_started.html)  I need to run the following command: `GO111MODULE=on go get github.com/quay/clair/v4/cmd/clairctl@latest` .

The command is not working I get the message: `go: package github.com/quay/clair/v4/cmd/clairctl is in the main module, so can't request version latest`. Then I tried : `GO111MODULE=on go install github.com/quay/clair/v4/cmd/clairctl@latest`. 

But after that I still can't use clairctl. 

I would really appreciate your help.

**Comment by hdonnay** [comment, marked as answer]
How do you mean "can't use?"

**Reply by inf3790** [reply]
The command simply isn't found. Here my output: 
`labadmin@cc-coskun2:~/test/clair$ GO111MODULE=on go install github.com/quay/clair/v4/cmd/clairctl@latest
labadmin@cc-coskun2:~/test/clair$ clairctl report ubuntu:focal
clairctl: command not found`

**Reply by hdonnay** [reply]
It was placed where ever you have the go install configured to put it. Look in the directory reported by `go env GOBIN`.

**Reply by inf3790** [reply]
Thanks that worked, now I'm at the next step. How do I scan an image. What do I need to do before that. For example I did: `docker pull ubuntu:latest` and after that I did `$ ./clairctl report ubuntu` . My output is: `2022-12-14T16:04:33Z ERR error="unexpected return status: 401"`.

**Reply by hdonnay** [reply]
Make sure that you're using the `config` and `host` flags as needed -- see `clairctl help` for the online help.

**Reply by hima-Baghel** [reply]
@inf3790 were you able to install and use clair I am facing the same issue as yours could you please help me out.
